src/parser/esprima_cpp/esprima.cpp

```diff
@@ -5248,6 +5248,10 @@ class Parser : public gc {
 
     PassRefPtr<Node> parseFunctionSourceElements()
     {
+        RefPtr<StatementContainer> statementBodyBefore = this->context->statementBody;
+        RefPtr<StatementContainer> body = StatementContainer::create();
+        this->context->statementBody = body;
```

Comment:
Why does `context` keep the reference of body node? To reparse the parameter lists? How about just adding default parameter nodes in advance? Could you explain it specifically?

Reply:
The reason behind that the ProgramNode starts the bytecode generation by its statement list step by step. But the statement list is not exist before the real function body parsing has not started, but the default arguments should be added to this list somehow. But I like your suggestion probably worth a try.